rules/CodeQuality/Rector/Foreach_/ForeachToInArrayRector.php

```diff
@@ -157,10 +157,27 @@ private function shouldSkipForeach(Foreach_ $foreach): bool
             return true;
         }
 
-        if (! $foreach->stmts[0] instanceof If_) {
+        $ifStatement = $foreach->stmts[0];
+
+        if (! $ifStatement instanceof If_) {
             return true;
         }
 
+        $ifCondition = $ifStatement->cond;
+
+        if ($ifCondition instanceof Expr\BinaryOp && (
+            $ifCondition->left instanceof FuncCall
+            || $ifCondition->left instanceof Expr\MethodCall
+            || $ifCondition->left instanceof Expr\NullsafeMethodCall
+            || $ifCondition->left instanceof Expr\StaticCall
+            || $ifCondition->right instanceof FuncCall
+            || $ifCondition->right instanceof Expr\MethodCall
+            || $ifCondition->right instanceof Expr\NullsafeMethodCall
+            || $ifCondition->right instanceof Expr\StaticCall
```

Comment:
you may can check when foreach variable value is used instead, that may can be checked by line 104, because this change should be valid on outer value:

```diff
       $outervalue = 'value';
-       foreach ($items as $item) {
-            if ($item == strtoupper($outervalue)) {
-                return true;
-            }
-        }
-        return false;
+        return in_array(strtoupper($outervalue), $items);
```